dp/matrix_chain_order.py

```diff
@@ -0,0 +1,46 @@
+import sys
+'''
+Dynamic Programming
+Implementation of Matrix Chain Multiplication
+Time Complexity: O(n^3)
+Space Complexity: O(n^2)
+'''
+def MatrixChainOrder(array):
+    N=len(array)
+    Matrix=[[0 for x in range(N)] for x in range(N)]
+    Sol=[[0 for x in range(N)] for x in range(N)]
+    for i in range(1,N):
+        Matrix[i][i]=0
```

Comment:
The loop
```python
for i in range(1,N):
       Matrix[i][i]=0
```
is redundant. 
Also the declaration of Matrix  and Sol can be reduced to
```python
Matrix = [[0] * n] * n
Sol = [[0] * n] * n
```